plugin/evm/evm_call_test.go

```diff
@@ -0,0 +1,95 @@
+
+	parsed, err := abi.JSON(strings.NewReader(contractABI))
+	require.NoError(t, err)
+
+	contractAddr, _, _, err := bind.DeployContract(auth, parsed, common.FromHex(contractBin), be, "v0")
+	require.NoError(t, err)
+
+	be.Commit(true)
+
+	contract := EVMContract{
+		address: contractAddr,
+		abi:     parsed,
+	}
+
+	blockChain := be.Blockchain()
+
+	evmBind := BindEVM(blockChain)
+
+	t.Run("nominal case", func(t *testing.T) {
+		header := blockChain.GetHeaderByNumber(1)
+
+		ret := evmBind.AtBlock(header.Hash()).OnContract(contract).Execute("setItem", [32]byte{1}, [32]byte{1})
+		require.NoError(t, ret.err)
+
+		be.Commit(true)
+
+		header = blockChain.GetHeaderByNumber(2)
+
+		ret = evmBind.AtBlock(header.Hash()).OnContract(contract).Execute("items", [32]byte{1})
+		require.NoError(t, ret.err)
+		require.Len(t, ret.values, 1)
```

Comment:
I'm missing something here that checks that the value was actually correctly set on the smart contract? We just check there is a return value, but not its actual value.

Reply:
Since we are only reading from smart contracts and not writing, I guess I'll need some extra code to actually write something to the smart contract. Correct me if I'm wrong.

P.S. I've tested it using standard eth libs to generate the smart contract, send some eth there and it worked fine. But here it just doesn't work :(